src/PrestaShopBundle/Twig/Layout/LayoutBuilder.php

```diff
@@ -0,0 +1,188 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Twig\Layout;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Entity\Tab;
+use PrestaShopBundle\Service\DataProvider\UserProvider;
+use Shop;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Tools;
+
+class LayoutBuilder
+{
+    private ?string $displayBackOfficeTop = null;
+
+    public function __construct(
+        private readonly LegacyContext $context,
+        private readonly bool $debugMode,
+        private readonly RequestStack $requestStack,
+        private readonly CsrfTokenManagerInterface $tokenManager,
+        private readonly UserProvider $userProvider,
+        private readonly string $psVersion,
+        private readonly Configuration $configuration,
+        private readonly HookDispatcherInterface $hookDispatcher,
+        private readonly MenuBuilder $menuBuilder,
+        private readonly TabRepository $tabRepository,
+        private readonly LocaleRepository $localeRepository,
+    ) {
+    }
+
+    /**
+     * Used to set the current locale in the context.
+     * TODO: Need to be removed when we rework on contexts.
+     */
+    public function setCurrentLocale(): void
+    {
+        $this->context->getContext()->currentLocale = $this->localeRepository->getLocale(
+            $this->context->getLanguage()->getLocale()
+        );
+    }
+
+    /**
+     * Enable New theme for smarty to avoid some problems with kpis for instance...
+     * TODO: Need to be removed when we will render kpis with twig only!
+     */
+    public function enableSmartyNewTheme(): void
+    {
+        $this->context->getContext()->smarty->setTemplateDir(_PS_BO_ALL_THEMES_DIR_ . 'new-theme/template/');
+    }
+
+    public function getIsoUser(): string
+    {
+        return $this->context->getLanguage()->getIsoCode();
+    }
+
+    public function isRtlLanguage(): bool
+    {
+        return (bool) $this->context->getLanguage()->isRTL();
+    }
+
+    public function getControllerName(): string
+    {
+        return htmlentities(Tools::getValue('controller'));
+    }
+
+    public function isMultiShop(): bool
+    {
+        return Shop::isFeatureActive();
+    }
+
+    public function isCollapseMenu(): bool
```

Comment:
What does it mean? 😅 not sure the prefix `is` is adapted, the layout is not a collapsed menu, maybe instead?

```suggestion
    public function isMenuCollapsed(): bool
```